src/main/java/uk/gov/hmcts/reform/translate/security/filter/CustomPermitAllFilter.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.translate.security.filter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import uk.gov.hmcts.reform.translate.model.ControllerConstants;
+import uk.gov.hmcts.reform.translate.security.CustomPermitAllAuthenticationTokenBuilder;
+import uk.gov.hmcts.reform.translate.security.SecurityUtils;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static uk.gov.hmcts.reform.translate.security.SecurityUtils.SERVICE_AUTHORIZATION;
+
+@Slf4j
+@Component
+public class CustomPermitAllFilter extends OncePerRequestFilter {
+
+    private final SecurityUtils securityUtils;
+
+    private static final String PUT_METHOD = "PUT";
+
+    @Autowired
+    public CustomPermitAllFilter(final SecurityUtils securityUtils) {
+        this.securityUtils = securityUtils;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+
+        if (isPutDictionaryEndpoint(request) && isPermittedService(request)) {
+            final AbstractAuthenticationToken authenticationToken =
+                new CustomPermitAllAuthenticationTokenBuilder(request)
+                    .build();
+
+            SecurityContextHolder.getContext().setAuthentication(authenticationToken);
+            filterChain.doFilter(request, response);
+            return;
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private boolean isPutDictionaryEndpoint(final HttpServletRequest request) {
+        return PUT_METHOD.equalsIgnoreCase(request.getMethod())
+            && ControllerConstants.DICTIONARY_URL.equalsIgnoreCase(request.getServletPath());
+    }
+
+    private boolean isPermittedService(final HttpServletRequest request) {
+        final String bearerToken = extractBearerToken(request);
+        final String serviceName = securityUtils.getServiceNameFromS2SToken(bearerToken);
```

Comment:
Can just pass in `request.getHeader(SERVICE_AUTHORIZATION)` to `getServiceNameFromS2SToken`, as that method removes the BEARER prefix from the token anyway....can then get rid of `extractBearerToken` method

Reply:
Fixed in 2613f5856df83f8634ca449585342333779c169d